drivers/display/display_gc9a01a.c

```diff
@@ -0,0 +1,599 @@
+/**
+ * Copyright (c) 2017 Jan Van Winkel <jan.van_winkel@dxplore.eu>
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ * Copyright (c) 2020 Teslabs Engineering S.L.
+ * Copyright (c) 2021 Krivorot Oleg <krivorot.oleg@gmail.com>
+ * Copyright (c) 2023 Mr Beam Lasers GmbH.
+ * Copyright (c) 2023 Amrith Venkat Kesavamoorthi <amrith@mr-beam.org>
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * @see https://www.buydisplay.com/download/ic/GC9A01A.pdf
+ */
+
+#include "display_gc9a01a.h"
+
+#include <zephyr/dt-bindings/display/gc9a01a.h>
+#include <zephyr/drivers/display.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(display_gc9a01a, CONFIG_DISPLAY_LOG_LEVEL);
+
+/*Display data struct*/
+struct gc9a01a_data {
+	uint8_t bytes_per_pixel;
+	enum display_pixel_format pixel_format;
+	enum display_orientation orientation;
+};
+
+/**
+ * @brief Initialize GC9A01A registers with DT values.
+ *
+ * @param dev GC9A01A device instance.
+ * @return 0 on success, errno otherwise.
+ */
+int gc9a01a_regs_init(const struct device *dev)
+{
+
+	const struct gc9a01a_config *config = dev->config;
+	const struct gc9a01a_regs *regs = config->regs;
+
+	// https://github.com/jakkra/ZSWatch init logic updated from this repo
+	const uint8_t *addr = regs->reg_arr;
+	uint8_t cmd, x, numArgs;
+	while ((cmd = *addr++) > 0) {
+		x = *addr++;
+		numArgs = x & 0x7F;
+		gc9a01a_transmit(dev, cmd, addr, numArgs);
+		addr += numArgs;
+		if (x & 0x80) {
+			k_msleep(GC9A01A_SLEEP_OUT_TIME +
+				 30); // 30ms on top of the 120ms sleepout time to account for any
+				      // manufacturing defects.
+		}
+	}
+	return 0;
+}
+
+/**
+ * @brief Transmit values to the display driver
+ *
+ * @param dev GC9A01A device instance.
+ * @param cmd command associated with the register.
+ * @param tx_data Data to be sent.
+ * @param tx_len Length of buffer to be sent.
+ * @return 0 on success, errno otherwise.
+ */
+int gc9a01a_transmit(const struct device *dev, uint8_t cmd, const void *tx_data, size_t tx_len)
+{
+	const struct gc9a01a_config *config = dev->config;
+
+	int r;
+	struct spi_buf tx_buf;
+	struct spi_buf_set tx_bufs = {.buffers = &tx_buf, .count = 1U};
+
+	/* send command */
+	tx_buf.buf = &cmd;
+	tx_buf.len = 1U;
+
+	gpio_pin_set_dt(&config->cmd_data, GC9A01A_CMD);
```

Comment:
Maybe also checking the return value here?

Reply:
added